qiskit/circuit/equivalence.py

```diff
@@ -98,16 +169,14 @@ def set_entry(self, gate, entry):
             entry (List['QuantumCircuit']) : A list of QuantumCircuits, each
                 equivalently implementing the given Gate.
         """
-
         for equiv in entry:
             _raise_if_shape_mismatch(gate, equiv)
             _raise_if_param_mismatch(gate.params, equiv.parameters)
 
         key = Key(name=gate.name, num_qubits=gate.num_qubits)
-
         equivs = [Equivalence(params=gate.params.copy(), circuit=equiv.copy()) for equiv in entry]
 
-        self._map[key] = Entry(search_base=False, equivalences=equivs)
+        self._graph[self._set_default_node(key)]["equivs"] = equivs
```

Comment:
Calling this function invalidates the property `num_gates_for_rule`, but since that property is keyed by insertion order, and there's no easy way to retrieve that for a given key, it's going to be pretty hard to make sure this updates it.

Reply:
Would be fixed by the above change, I believe.